Renata/synthetic population (#15)
AequilibraE/tradesman: docs/source/synthetic_population.rst

female_7,TAZ,persons,100000,POPF7,(persons.AGEP >= 25) & (persons.AGEP < 30) & (persons.SEX == 2)
female_8,TAZ,persons,100000,POPF8,(persons.AGEP >= 30) & (persons.AGEP < 35) & (persons.SEX == 2)
female_9,TAZ,persons,100000,POPF9,(persons.AGEP >= 35) & (persons.AGEP < 40) & (persons.SEX == 2)
female_10,TAZ,persons,100000,POPF10,(persons.AGEP >= 40) & (persons.AGEP < 45) & (persons.SEX == 2)
female_11,TAZ,persons,100000,POPF11,(persons.AGEP >= 45) & (persons.AGEP < 50) & (persons.SEX == 2)
female_12,TAZ,persons,100000,POPF12,(persons.AGEP >= 50) & (persons.AGEP < 55) & (persons.SEX == 2)
female_13,TAZ,persons,100000,POPF13,(persons.AGEP >= 55) & (persons.AGEP < 60) & (persons.SEX == 2)
female_14,TAZ,persons,100000,POPF14,(persons.AGEP >= 60) & (persons.AGEP < 65) & (persons.SEX == 2)
female_15,TAZ,persons,100000,POPF15,(persons.AGEP >= 65) & (persons.AGEP < 70) & (persons.SEX == 2)
female_16,TAZ,persons,100000,POPF16,(persons.AGEP >= 70) & (persons.AGEP < 75) & (persons.SEX == 2)
female_17,TAZ,persons,100000,POPF17,(persons.AGEP >= 75) & (persons.AGEP < 80) & (persons.SEX == 2)
female_18,TAZ,persons,100000,POPF18,(persons.AGEP >= 80) & (persons.SEX == 2)

> For each sex and age structure, we create a control variable, considering the geography level of TAZ. For both women and men, we have 18 variables, named POPF and POPM, for female and male inhabitants. For TAZs we also consider the expected number of households. In the following section, we discuss the premisses of the model, and explain how the expected number of households was calculated. Finally, we have the controls by REGION, an upper level. For regional control, we consider the expected number of households of different sizes.

Reviewer: ```suggestion
For each sex and age bracket, we create a control variable, considering the geography level of TAZ. For both women and men, we have 18 variables, named POPF and POPM, for female and male residents. For TAZs we also consider the expected number of households. 

In the following section, we discuss the premisses of the model and explain how the expected number of households was calculated. Finally, we have the controls by REGION, an upper level. For regional control, we consider the expected number of households of different sizes.
```

> Tradesman model premisses

Reviewer: ```suggestion
Tradesman model premises
```

~~~~~~~~~~~~~~~~~~~~~~~~~

> PopulationSim library requires a control file to run its processes. In the control file, one can specify and check the requirements for the synthetic population. The only mandatory requirement for a PopulationSim is generating the right number of households in each geography and this specification is made at the lowest level. The lowest level in Tradesman is the TAZs (Traffic Assessment Zones), however, the data necessary to estimate the total households in the upper level. 

Reviewer: ```suggestion
The PopulationSim library requires a control file to run. In the control file, one can specify and check the requirements for the synthetic population. The only mandatory requirement for a PopulationSim is generating the right number of households in each geography and this specification is made at the lowest level. The lowest level in an AequilibraE model is the TAZs (Traffic Analysis Zones), however, the data necessary to estimate the total households in the upper level. 
```

I am not quite sure what do you mean by the last sentence

> To fulfill the TAZ requirement, we estimated the total households based on the United Nations `Population Division <https://www.un.org/development/desa/pd/data/household-size-and-composition>`_ information about household composition. As there is no data available for all countries or autonomous regions, for countries that are not present in the sample, we considered their household informations to be similar to their closest neighbors. As there are many data sources available in the Population Division sheet, we kept the most recent data for each country, as long as it had information about the total number of households.

Reviewer: ```suggestion
To fulfill the requirement of specifying the number of households at the TAZ level, we estimated the total households based on the United Nations `Population Division <https://www.un.org/development/desa/pd/data/household-size-and-composition>`_ information about household composition, with the underlying hypothesis that the family size distribution would be equal for all TAZs in the country. 

As there is no data available for all countries or autonomous regions, for countries that are not present in the sample, we considered their household information to be similar to their closest (and most similar) neighbors. As there are many data sources available in the Population Division sheet, we kept the most recent data for each country, as long as it had information about the total number of households.
```

> The total number of households in each TAZ is a sum of all households with different sizes. To match the information provided by the UN, we consider households with 1 person (HHBASE1), 2 to 3 persons (HHBASE2), 4 to 5 persons (HHBASE4), and with more than 6 persons (HHBASE6). Hence, we sum all population by sex and age in the zone, divide by the average number of persons in the household, and multiply by the percentage of households of each size expected for the country. Finally, we sum the integer expected values for each household size to obtain the mandatory expected number of households in the TAZ (HHBASE).

Reviewer: What number did you use for the average size of families with 6 or more people?  It is necessarily larger than 6...

- Percentage of households with at least one person under 15 years old;
- Percentage of households with at least one person under 18 years old;
- Percentage of households with at least one person under 20 years old;
- Percentage of households with at least one person over 60 years old;
- Percentage of households with at least one person over 65 years old;
- Percentage of households with at least one person under 15 and one person over 60;
- Percentage of households with at least one person under 15 and one person over 65;
- Percentage of households with at least one person under 18 and one person over 60;
- Percentage of households with at least one person under 18 and one person over 65;
- Percentage of households with at least one person under 20 and one person over 60;
- Percentage of households with at least one person under 20 and one person over 65.

As part of the validation process, we also run the validation process presented by the development team of PopulationSim, available `here <https://github.com/activitysim/populationSim/tree/master/scripts>`_.

> 

Reviewer: I think it would be helpful to include images of the validations we produce and mention that they are produced alongside with the results (I didn;t see that, but we should create the validation charts as images alongside the results on disk

Reviewer: I would put the link to the notebook itself https://github.com/ActivitySim/populationsim/blob/master/scripts/validation.ipynb
